Project.Xml.Test/XmlTest.cs

```diff
@@ -0,0 +1,203 @@
+        }
+
+        [TestMethod]
+        public void File_WithInvalidXMLFormat()
+        {
+            //arrange
+            IFileHandling ifh = new XmlParser();
+
+            //assert
+            Assert.ThrowsException<XmlException>(() => ifh.GetParsedData("<xml>, <orange id = 'Round Orange' >, </xml>"));
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLNonNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <orange id = 'Round Orange' /><orange id = 'Naval Orange' /><orange id = 'Blood Orange' /><orange /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "Blood Orange", "Naval Orange", "Round Orange", null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
+
+            //assert
+            //validate that the sorted testresults match the sorted expected results
+            for (int i = 0; i < expectedResults.Length; i++)
+            {
+                Assert.AreEqual(testResults[i], expectedResults[i]);
+            }
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <age id = '55' /><age id = '92' /><age id = '25' /><orange /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "25", "55", "92", null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
+
+            //assert
+            //validate that the sorted testresults match the sorted expected results
+            for (int i = 0; i < expectedResults.Length; i++)
+            {
+                Assert.AreEqual(testResults[i], expectedResults[i]);
+            }
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLAlphaNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <orange id = 'Round Orange' /><orange id = 'Naval Orange' /><orange /><age id = '55' /><age id = '92' /><age /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "55", "92", "Naval Orange", "Round Orange", null, null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
```

Comment:
`Scope`: `GetSortedData` functionality is part of the console (not the parser) and should be tested in `Project.ConsoleApp.Test` via `Program.Main()`.